### ISSUE TITLE
SchedulerPredicates limits number of pods failing on upgraded cluster (1.1->1.2)

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#23018]
**SchedulerPredicates validates resource limits of pods that are allowed to run [Conformance]**

*/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:348 Expected : 17 to equal : 20*

```
14:27:58 STEP: Starting additional 10 Pods to fully saturate the cluster CPU and trying to start another one
14:27:59 Mar 15 14:27:59.422: INFO: 10 pods running
14:28:04 Mar 15 14:28:04.476: INFO: 18 pods running
14:28:09 Mar 15 14:28:09.495: INFO: 18 pods running
14:28:14 Mar 15 14:28:14.516: INFO: 18 pods running
14:28:19 Mar 15 14:28:19.536: INFO: 18 pods running
14:28:24 Mar 15 14:28:24.554: INFO: 18 pods running
14:28:29 Mar 15 14:28:29.596: INFO: 18 pods running
14:28:34 Mar 15 14:28:34.617: INFO: 18 pods running
14:28:39 Mar 15 14:28:39.637: INFO: 18 pods running
14:28:44 Mar 15 14:28:44.677: INFO: 18 pods running
14:28:49 Mar 15 14:28:49.702: INFO: 18 pods running
14:28:54 Mar 15 14:28:54.724: INFO: 18 pods running
14:28:59 Mar 15 14:28:59.743: INFO: 18 pods running
14:29:04 Mar 15 14:29:04.769: INFO: 18 pods running
14:29:09 Mar 15 14:29:09.787: INFO: 18 pods running
14:29:14 Mar 15 14:29:14.807: INFO: 18 pods running
14:29:19 Mar 15 14:29:19.830: INFO: 18 pods running
14:29:24 Mar 15 14:29:24.849: INFO: 18 pods running
14:29:29 Mar 15 14:29:29.871: INFO: 18 pods running
14:29:34 Mar 15 14:29:34.890: INFO: 18 pods running
14:29:39 Mar 15 14:29:39.908: INFO: 18 pods running
14:29:44 Mar 15 14:29:44.933: INFO: 18 pods running
14:29:49 Mar 15 14:29:49.952: INFO: 18 pods running
14:29:54 Mar 15 14:29:54.977: INFO: 18 pods running
14:29:59 Mar 15 14:29:59.996: INFO: 18 pods running
14:30:05 Mar 15 14:30:05.015: INFO: 18 pods running
14:30:10 Mar 15 14:30:10.033: INFO: 18 pods running
14:30:15 Mar 15 14:30:15.050: INFO: 18 pods running
14:30:20 Mar 15 14:30:20.070: INFO: 18 pods running
14:30:25 Mar 15 14:30:25.088: INFO: 18 pods running
14:30:30 Mar 15 14:30:30.106: INFO: 18 pods running
14:30:35 Mar 15 14:30:35.126: INFO: 18 pods running
14:30:40 Mar 15 14:30:40.148: INFO: 18 pods running
14:30:45 Mar 15 14:30:45.165: INFO: 18 pods running
14:30:50 Mar 15 14:30:50.185: INFO: 18 pods running
14:30:55 Mar 15 14:30:55.208: INFO: 18 pods running
14:31:00 Mar 15 14:31:00.232: INFO: 18 pods running
14:31:05 Mar 15 14:31:05.255: INFO: 18 pods running
14:31:10 Mar 15 14:31:10.275: INFO: 18 pods running
14:31:15 Mar 15 14:31:15.297: INFO: 18 pods running
14:31:20 Mar 15 14:31:20.319: INFO: 18 pods running
14:31:25 Mar 15 14:31:25.344: INFO: 18 pods running
14:31:30 Mar 15 14:31:30.363: INFO: 18 pods running
14:31:35 Mar 15 14:31:35.386: INFO: 18 pods running
14:31:40 Mar 15 14:31:40.410: INFO: 18 pods running
14:31:45 Mar 15 14:31:45.427: INFO: 18 pods running
14:31:50 Mar 15 14:31:50.449: INFO: 18 pods running
14:31:55 Mar 15 14:31:55.473: INFO: 18 pods running
14:32:00 Mar 15 14:32:00.496: INFO: 18 pods running
14:32:05 Mar 15 14:32:05.513: INFO: 18 pods running
14:32:10 Mar 15 14:32:10.536: INFO: 18 pods running
14:32:15 Mar 15 14:32:15.562: INFO: 18 pods running
14:32:20 Mar 15 14:32:20.582: INFO: 18 pods running
14:32:25 Mar 15 14:32:25.606: INFO: 18 pods running
14:32:30 Mar 15 14:32:30.623: INFO: 18 pods running
14:32:35 Mar 15 14:32:35.641: INFO: 18 pods running
14:32:40 Mar 15 14:32:40.685: INFO: 18 pods running
14:32:45 Mar 15 14:32:45.706: INFO: 18 pods running
14:32:50 Mar 15 14:32:50.725: INFO: 18 pods running
14:32:55 Mar 15 14:32:55.745: INFO: 18 pods running
14:33:00 Mar 15 14:33:00.768: INFO: 18 pods running
14:33:05 Mar 15 14:33:05.790: INFO: 18 pods running
14:33:10 Mar 15 14:33:10.809: INFO: 18 pods running
14:33:15 Mar 15 14:33:15.839: INFO: 18 pods running
14:33:20 Mar 15 14:33:20.857: INFO: 18 pods running
14:33:25 Mar 15 14:33:25.875: INFO: 18 pods running
14:33:30 Mar 15 14:33:30.894: INFO: 18 pods running
14:33:35 Mar 15 14:33:35.915: INFO: 18 pods running
14:33:40 Mar 15 14:33:40.936: INFO: 18 pods running
14:33:45 Mar 15 14:33:45.957: INFO: 18 pods running
14:33:50 Mar 15 14:33:50.975: INFO: 18 pods running
14:33:55 Mar 15 14:33:55.995: INFO: 18 pods running
14:34:01 Mar 15 14:34:01.016: INFO: 18 pods running
14:34:06 Mar 15 14:34:06.035: INFO: 18 pods running
14:34:11 Mar 15 14:34:11.057: INFO: 18 pods running
14:34:16 Mar 15 14:34:16.076: INFO: 18 pods running
14:34:21 Mar 15 14:34:21.099: INFO: 18 pods running
14:34:26 Mar 15 14:34:26.118: INFO: 18 pods running
14:34:31 Mar 15 14:34:31.139: INFO: 18 pods running
14:34:36 Mar 15 14:34:36.158: INFO: 18 pods running
14:34:41 Mar 15 14:34:41.178: INFO: 18 pods running
14:34:46 Mar 15 14:34:46.197: INFO: 18 pods running
14:34:51 Mar 15 14:34:51.218: INFO: 18 pods running
14:34:56 Mar 15 14:34:56.236: INFO: 18 pods running
14:35:01 Mar 15 14:35:01.256: INFO: 18 pods running
14:35:06 Mar 15 14:35:06.278: INFO: 18 pods running
14:35:11 Mar 15 14:35:11.300: INFO: 18 pods running
14:35:16 Mar 15 14:35:16.321: INFO: 18 pods running
14:35:21 Mar 15 14:35:21.342: INFO: 18 pods running
14:35:26 Mar 15 14:35:26.362: INFO: 18 pods running
14:35:31 Mar 15 14:35:31.387: INFO: 18 pods running
14:35:36 Mar 15 14:35:36.405: INFO: 18 pods running
14:35:41 Mar 15 14:35:41.424: INFO: 18 pods running
14:35:46 Mar 15 14:35:46.443: INFO: 18 pods running
14:35:51 Mar 15 14:35:51.461: INFO: 18 pods running
14:35:56 Mar 15 14:35:56.480: INFO: 18 pods running
14:36:01 Mar 15 14:36:01.502: INFO: 18 pods running
14:36:06 Mar 15 14:36:06.520: INFO: 18 pods running
14:36:11 Mar 15 14:36:11.540: INFO: 18 pods running
14:36:16 Mar 15 14:36:16.560: INFO: 18 pods running
14:36:21 Mar 15 14:36:21.583: INFO: 18 pods running
14:36:26 Mar 15 14:36:26.601: INFO: 18 pods running
14:36:31 Mar 15 14:36:31.620: INFO: 18 pods running
14:36:36 Mar 15 14:36:36.649: INFO: 18 pods running
14:36:41 Mar 15 14:36:41.671: INFO: 18 pods running
14:36:46 Mar 15 14:36:46.688: INFO: 18 pods running
14:36:51 Mar 15 14:36:51.707: INFO: 18 pods running
14:36:56 Mar 15 14:36:56.726: INFO: 18 pods running
14:37:01 Mar 15 14:37:01.747: INFO: 18 pods running
14:37:06 Mar 15 14:37:06.767: INFO: 18 pods running
14:37:11 Mar 15 14:37:11.792: INFO: 18 pods running
14:37:16 Mar 15 14:37:16.809: INFO: 18 pods running
14:37:21 Mar 15 14:37:21.839: INFO: 18 pods running
14:37:26 Mar 15 14:37:26.858: INFO: 18 pods running
14:37:31 Mar 15 14:37:31.877: INFO: 18 pods running
14:37:36 Mar 15 14:37:36.894: INFO: 18 pods running
14:37:41 Mar 15 14:37:41.917: INFO: 18 pods running
14:37:46 Mar 15 14:37:46.937: INFO: 18 pods running
14:37:51 Mar 15 14:37:51.959: INFO: 18 pods running
14:37:56 Mar 15 14:37:56.979: INFO: 18 pods running
14:38:01 Mar 15 14:38:01.999: INFO: 18 pods running
14:38:01 Mar 15 14:38:01.999: INFO: Timed out after 10m0s waiting for pods to start running.
14:38:02 [AfterEach] SchedulerPredicates
14:38:02   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:192
14:38:02 STEP: Destroying namespace for this suite e2e-tests-sched-pred-xcyz6
14:38:22 
14:38:22 • Failure [625.412 seconds]
14:38:22 SchedulerPredicates
14:38:22 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:389
14:38:22   validates resource limits of pods that are allowed to run [Conformance] [It]
14:38:22   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:348
14:38:22 
14:38:22   Expected
14:38:22       <int>: 18
14:38:22   to equal
14:38:22       <int>: 20
14:38:22 
14:38:22   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:87
```

View logs and artifacts at https://storage.cloud.google.com/kubernetes-jenkins/logs/kubernetes-upgrade-gke-1.1-1.2-step6-e2e-old/22

**Comment by ihmccreery** [comment]
This is v1.1 tests running against a cluster with both master & nodes upgraded to v1.2.

**Comment by ihmccreery** [comment]
@davidopp to triage

**Comment by ihmccreery** [comment]
I'm also looking into it.

**Comment by ihmccreery** [comment]
It looks like the test isn't registering the resource requests by some of the pods (namely fluentd and kubeproxy):

```
Mar 15 14:27:58.639: INFO: Pod fluentd-cloud-logging-gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg requesting capacity 0 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg
Mar 15 14:27:58.639: INFO: Pod fluentd-cloud-logging-gke-upgrade-gke-1-1-1-2-4dc5394b-node-wc89 requesting capacity 0 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-wc89
Mar 15 14:27:58.639: INFO: Pod fluentd-cloud-logging-gke-upgrade-gke-1-1-1-2-4dc5394b-node-yyd7 requesting capacity 0 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-yyd7
Mar 15 14:27:58.639: INFO: Pod heapster-v1.0.0-beta1-064dl requesting capacity 100 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg
Mar 15 14:27:58.639: INFO: Pod kube-dns-v11-p5o2r requesting capacity 310 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg
Mar 15 14:27:58.639: INFO: Pod kube-proxy-gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg requesting capacity 0 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg
Mar 15 14:27:58.639: INFO: Pod kube-proxy-gke-upgrade-gke-1-1-1-2-4dc5394b-node-wc89 requesting capacity 0 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-wc89
Mar 15 14:27:58.639: INFO: Pod kube-proxy-gke-upgrade-gke-1-1-1-2-4dc5394b-node-yyd7 requesting capacity 0 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-yyd7
Mar 15 14:27:58.639: INFO: Pod kubernetes-dashboard-v1.0.0-8smd9 requesting capacity 100 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg
Mar 15 14:27:58.639: INFO: Pod l7-lb-controller-v0.6.0-u8dby requesting capacity 110 on Node gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg
Mar 15 14:27:58.639: INFO: Node: gke-upgrade-gke-1-1-1-2-4dc5394b-node-yyd7 has capacity: 2000
Mar 15 14:27:58.639: INFO: Node: gke-upgrade-gke-1-1-1-2-4dc5394b-node-cqxg has capacity: 1380
Mar 15 14:27:58.639: INFO: Node: gke-upgrade-gke-1-1-1-2-4dc5394b-node-wc89 has capacity: 2000
```

**Comment by ihmccreery** [comment]
It looks like this is because https://github.com/kubernetes/kubernetes/pull/18218 was never backported to 1.1.  I'm bumping this down to P1, but I'd like to get it fixed as soon as possible regardless.

**Comment by bgrant0607** [comment]
Great catch, @ihmccreery!

**Comment by ihmccreery** [comment]
Fixed in #23029.
